### PR TITLE
Add hls playable field and notification

### PR DESCRIPTION
**Pull request by Karolk99**

### CODE REVIEW

File: test/jellyfish_web/integration/server_socket_test.exs [hunk 1]
```diff
@@ -175,6 +184,21 @@ defmodule JellyfishWeb.Integration.ServerSocketTest do
...
 
+    correct_components? =
+      components
+      |> Enum.map(fn %Component{component: component} -> component end)
+      |> Enum.all?(fn
+        {:hls, %Hls{playable: false}} -> true
+        {:rtsp, %Rtsp{}} -> true
+      end)
+
+    assert correct_components?
```
Comment (Rados13) [inline test/jellyfish_web/integration/server_socket_test.exs:195]:
```suggestion
    assert true =
      components
      |> Enum.map(fn %Component{component: component} -> component end)
      |> Enum.all?(fn
        {:hls, %Hls{playable: false}} -> true
        {:rtsp, %Rtsp{}} -> true
        _other -> false
      end)

```

File: lib/jellyfish/room.ex [hunk 2]
```diff
@@ -321,6 +322,25 @@ defmodule Jellyfish.Room do
...
+      end)
+
+    Phoenix.PubSub.broadcast(
+      Jellyfish.PubSub,
+      "server_notification",
+      {:hls_playable, state.id, endpoint_id}
+    )
+
+    {:noreply, state}
+  end
```
Comment (Rados13) [inline lib/jellyfish/room.ex:342]: Shouldn't we store this information somewhere?

File: lib/jellyfish/component.ex [hunk 3]
```diff
@@ -44,14 +48,20 @@ defmodule Jellyfish.Component do
   @spec new(component(), map()) :: {:ok, t()} | {:error, term()}
   def new(type, options) do
     with {:ok, endpoint} <- type.config(options) do
+      metadata = get_metadata(type)
```
Comment (roznawsk) [inline lib/jellyfish/component.ex:51]: Perhaps we could move the logic of getting metadata to the modules in question?
```suggestion
      metadata = type.metadata()
```

File: lib/jellyfish/room.ex (hunk 2 above)
Comment (roznawsk) [reply lib/jellyfish/room.ex:342]: I think it should be stored inside the HLS component

**Review by roznawsk — CHANGES_REQUESTED** [review]

**Review by LVala — APPROVED** [review]
👍

**Pull request merged**